server.go

```diff
@@ -348,7 +349,20 @@ func NewFromConfig(logger *logrus.Logger, conf Config) (*Server, error) {
 			}
 
 			ret.spanSinks = append(ret.spanSinks, ddSink)
-			logger.Info("Configured Datadog trace sink")
+			logger.Info("Configured Datadog span sink")
+		}
+
+		if conf.XrayAddress != "" {
+			if conf.XraySamplePercentage == 0 {
+				log.Warn("XRay sample percentage is 0, no segments will be sent.")
```

Comment:
Shouldn't we just disable the sink (with the warning) if the sample percentage is zero?